robot/EDA/tests/browser/settings/relationships_reciprocal.robot

```diff
@@ -0,0 +1,127 @@
+    ...         Female=Female1
+    ...         Male=Male1
+    ...         Neutral=Neutral1
+    Verify new setting checkbox status edit mode
+    ...         Active=false
+    Click action button on EDA settings page    Cancel
+    ${actual_count}=        Get total settings count
+    ${result}=        Evaluate        ${actual_count} > ${settings_count}
+    Log To Console          ${result}
+    # Below commented steps are alternate solution to verify count and let it be in test if needed
+    # ${no_of_records}=       Verify settings count       ${settings_count}       ${actual_count}
+    # ...                     expected=True
+    # Log To Console    ${no_of_records}
+    Enter new reciprocal setting
+    ...         Name=Name
+    ...         Female=Female
+    ...         Male=Male
+    ...         Neutral=Neutral
+    Set reciprocal setting status
+    ...         Active=true
+    Click add new setting button        New Reciprocal Setting      Add Setting
+    Scroll web page
+    ${final_count} =       Get total settings count
+    Should Be Equal As Integers     ${actual_count}     ${final_count}
+
+Verify reciprocal relationship settings can be deleted
+    [Documentation]         Validates reciprocal settings can be deleted. Also checks if those
+    ...                     values are retained after save.
+    [tags]                    unstable        W-8245483     rbt:high
+    Click action button on EDA settings page    Edit
+    Click delete setting icon       Name=Name1
+    Handle alert
+    Sleep       1
+    Verify setting deleted          True        Name=Name1
+    Click action button on EDA settings page    Save
+    Scroll web page
+    Verify setting removed          Name=Name1
+
+Verify existing reciprocal setting can be edited and values retained on save
+    [Documentation]         Validates reciprocal settings can be updated. Also checks if those
+    ...                     values are retained after save.
+    [tags]                    unstable        W-8251460     rbt:high
+    Click action button on EDA settings page    Edit
+    Update existing reciprocal setting      column=Name1
+    ...                                     Name=Name1
+    ...                                     Female=Female1
+    ...                                     Male=Male1
+    ...                                     Neutral=Neutral1
+    Set reciprocal setting status
+    ...         Active=true
+    Click action button on EDA settings page    Save
+    Scroll web page
+    Verify updated reciprocal setting      column=Name1
+    ...                                    Name=Name1
+    ...                                    Female=Female1
+    ...                                    Male=Male1
+    ...                                    Neutral=Neutral1
+
+Verify reciprocal relationship settings can be deleted to clean data
+    [Documentation]         Validates reciprocal settings can be deleted. Also checks if those
```

Comment:
Do you want to call this in Teardown instead ? I am sure you thought about it and had some challenges. Just trying to understand the reasoning

Reply:
I thought keeping more than 5 steps in teardown is not a best way to do it and added as a separate test which looks good. If we had the ability to recall/reuse a test in robot then I would have included it in teardown.